### ISSUE TITLE
[Bug] override url with configurator from "service-name.configurators" in interface mode  not as expected

### ISSUE THREAD

**Issue by luoning810** [apache/dubbo#14185]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dubbo/issues?q=is%3Aissue) and found no similar issues.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Dubbo Version

Dubbo java 3.2.11，oracle jdk 1.8

##### step 1  add service mode dynamic config
<img width="786" alt="企业微信截图_5debfbe8-a741-4fdd-88d5-8645874aac09" src="https://github.com/apache/dubbo/assets/22349611/9457540e-137c-406b-b926-3c6fa796e16b">

##### step 2 ReferenceConfigurationListener#notifyOverrides
<img width="1109" alt="企业微信截图_9de645f3-0b88-4731-8d2b-4cca5fc8e1d1" src="https://github.com/apache/dubbo/assets/22349611/3b566846-2c74-4db7-b385-1dd41dabbc32">

##### step4 mergeUrl
<img width="1289" alt="企业微信截图_15691103-91b4-4997-baae-44d8e6be9d21" src="https://github.com/apache/dubbo/assets/22349611/08861291-d3c8-40ba-ba77-e499473b7bec">

##### step5 overrideWithReferenceConfigurators  
<img width="1417" alt="企业微信截图_4bfa068e-a653-4438-8751-152570f26646" src="https://github.com/apache/dubbo/assets/22349611/9eb2f3f0-8f68-4290-9c86-0e6f4d41d880">

##### step6 service configurator rule mismatch
<img width="1474" alt="企业微信截图_87b5de32-7f28-479a-836d-bcdba3fe75ba" src="https://github.com/apache/dubbo/assets/22349611/5a505bee-02bb-492e-8384-812bce57e97b">

### What you expected to happen

isV27ConditionMatchOrUnset return true when application  and  service is matched

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
@chickenlj PTAL

**Comment by laywin** [comment]
currently, when update dynamic configuration in dubbo-admin,  the key is only serviceName, no group and version to be registered in registry center. so it can't be matched. temporarily, You can manually add group and version to the dynamic configuration to solve this problem. just like this:

```java

configVersion: v2.7
enabled: true
configs:
  - side: consumer
    addresses:
      - 0.0.0.0
    parameters:
      timeout: 2900
      group: test
      version: 1.0.0
```

I think it is best to modify the registration behavior of dubbo-admin.

**Comment by chickenlj** [comment]
@laywin From my understanding, it's an issue of dubbo-admin. dubbo-admin registered the rule with a wrong id or match key which caused the Dubbo SDK cannot recognize the rule, so it never gets parsed and worked on SDK. 

We realized the current dubbo-admin implementation has some issues especially with traffic management features. So we are working on building a brand new dubbo admin with Golang language. Aside from visualized console, we expect it to also support deploying Dubbo on Kubernetes by leveraging the Kubernetes Service concept.

The incubating repo is here https://github.com/apache/dubbo-kubernetes

**Comment by laywin** [comment]
> @laywin From my understanding, it's an issue of dubbo-admin. dubbo-admin registered the rule with a wrong id or match key which caused the Dubbo SDK cannot recognize the rule, so it never gets parsed and worked on SDK.
> 
> We realized the current dubbo-admin implementation has some issues especially with traffic management features. So we are working on building a brand new dubbo admin with Golang language. Aside from visualized console, we expect it to also support deploying Dubbo on Kubernetes by leveraging the Kubernetes Service concept.
> 
> The incubating repo is here https://github.com/apache/dubbo-kubernetes

sounds great!

**Comment by luoning810** [comment]
Thank you very much for the response above. We are currently validating the upgrade from Dubbo 2.7 to Dubbo 3.2. DubboAdmin is indeed using a version adapted to 2.7, but I understand that the differences in interface-level dynamic configuration between the old and new versions mainly lie in whether the key includes the group and version. Even if we remove the group and version from the key, Dubbo still does not work properly.

I tried to analyze the issue and believe the problem occurs in the following areas:

<img width="1199" alt="image" src="https://github.com/apache/dubbo/assets/22349611/b528648a-fa4a-4442-b3cc-0beeddb22c89">

When referenceConfigurationListener.getConfigurators() is not empty, the overriddenURL may need to add group and version parameters so that AbstractConfigurator#isV27ConditionMatchOrUnset(overriddenURL) returns the correct result when comparing the configServiceKey.

Currently, line 195 always returns false because the url is the overriddenURL mentioned above, which only includes the application name and does not contain the group and version.
<img width="1410" alt="image" src="https://github.com/apache/dubbo/assets/22349611/5aebc388-6386-4b57-ace2-084e6a2507f0">

**Comment by laywin** [comment]
This problem is that dubbo-admin don't registry group and version with key to the registration center.  you will see sdk has the operation to parse group and version in ConfigParser#appendService

```java
   private static String appendService(String serviceKey) {
        StringBuilder sb = new StringBuilder();
        if (StringUtils.isEmpty(serviceKey)) {
            throw new IllegalStateException("service field in configuration is null.");
        }

        String interfaceName = serviceKey;
        int i = interfaceName.indexOf('/');
        if (i > 0) {
            sb.append("group=");
            sb.append(interfaceName, 0, i);
            sb.append('&');

            interfaceName = interfaceName.substring(i + 1);
        }
        int j = interfaceName.indexOf(':');
        if (j > 0) {
            sb.append("version=");
            sb.append(interfaceName.substring(j + 1));
            sb.append('&');
            interfaceName = interfaceName.substring(0, j);
        }
        sb.insert(0, interfaceName + "?");

        return sb.toString();
    }
```

but serviceKey don't have group and version. you can also config them in dubbo admin 's parameters

```java
parameters:
      timeout: 2900
      group: test
      version: 1.0.0
```

**Comment by luoning810** [comment]
I agree with your point. The configurator URL of the override protocol can include the group and version added through DubboAdmin.
In the screenshot below, the group and version added by DubboAdmin are in the configuratorUrl, which I believe is as expected. However, the url variable does not contain the group and version, which I think is unexpected. This inconsistency leads to different results when calling getServiceKey with configuratorUrl compared to url.
![image](https://github.com/apache/dubbo/assets/22349611/0efc3dab-2464-48b1-ae91-60b51af2fee1)

**Comment by chickenlj** [comment]
> I agree with your point. The configurator URL of the override protocol can include the group and version added through DubboAdmin. In the screenshot below, the group and version added by DubboAdmin are in the configuratorUrl, which I believe is as expected. However, the url variable does not contain the group and version, which I think is unexpected. This inconsistency leads to different results when calling getServiceKey with configuratorUrl compared to url. ![image](https://private-user-images.githubusercontent.com/22349611/331065376-0efc3dab-2464-48b1-ae91-60b51af2fee1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MjkxOTcsIm5iZiI6MTcxNTkyODg5NywicGF0aCI6Ii8yMjM0OTYxMS8zMzEwNjUzNzYtMGVmYzNkYWItMjQ2NC00OGIxLWFlOTEtNjBiNTFhZjJmZWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE3VDA2NTQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzNGU2N2M2NTg2MGJlY2M4M2RiYWEwNTYxNzAzNjgwZTI5MjI1ZDhhOWQ2OTVjMjUxZTgzMGRhYjNiYzQyODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UaabLcIQveKgkzBoJV1h3euUoUK1qpRVXKiKkwa3bTo)

Thanks for the detailed explanation, it's indeed an matching issue, I will fix it now.

**Comment by chickenlj** [comment]
By the way, if you are using Dubbo 3.2.x, the following configuration rule is recommended:

```yaml
configVersion: v3.0
scope: service
key: org.apache.dubbo.samples.UserService
configs:
  - match:
      application:
        oneof:
          - exact: shop-frontend
    side: consumer
    parameters:
      retries: '4'
```

As explained here https://cn.dubbo.apache.org/zh-cn/overview/core-features/traffic/configuration-rule/

**Comment by luoning810** [comment]
Thank you for your response and suggestions.

**Comment by wcy666103** [comment]
Please close this pr as complete @AlbumenJ
